src/skmultiflow/meta/adaptive_random_forest_regressor.py

```diff
@@ -0,0 +1,497 @@
+from copy import deepcopy
+import math
+
+import numpy as np
+
+from sklearn.preprocessing import normalize
+
+from skmultiflow.core import BaseSKMObject, RegressorMixin, MetaEstimatorMixin
+from skmultiflow.drift_detection.base_drift_detector import BaseDriftDetector
+from skmultiflow.drift_detection import ADWIN
+from skmultiflow.trees.regression_hoeffding_tree import RegressionHoeffdingTree
+from skmultiflow.metrics.measure_collection import RegressionMeasurements
+from skmultiflow.utils import get_dimensions, normalize_values_in_dict, check_random_state, check_weights
+
+
+class AdaptiveRandomForestRegressor(BaseSKMObject, RegressorMixin, MetaEstimatorMixin):
+    """Adaptive Random Forest estimator.
+
+    Parameters
+    ----------
+    n_estimators: int, optional (default=10)
+        Number of trees in the ensemble.
+
+    max_features : int, float, string or None, optional (default="auto")
+        Max number of attributes for each node split.
+
+        - If int, then consider ``max_features`` features at each split.
+        - If float, then ``max_features`` is a percentage and
+          ``int(max_features * n_features)`` features are considered at
+          each split.
+        - If "auto", then ``max_features=sqrt(n_features)``.
+        - If "sqrt", then ``max_features=sqrt(n_features)`` (same as "auto").
+        - If "log2", then ``max_features=log2(n_features)``.
+        - If None, then ``max_features=n_features``.
+
+    disable_weighted_vote: bool, optional (default=False)
+        Weighted vote option.
+
+    lambda_value: int, optional (default=6)
+        The lambda value for bagging (lambda=6 corresponds to Leverage Bagging).
+
+    drift_detection_method: BaseDriftDetector or None, optional
+        (default=ADWIN(0.001))
+        Drift Detection method. Set to None to disable Drift detection.
+
+    warning_detection_method: BaseDriftDetector or None, default(ADWIN(0.01))
+        Warning Detection method. Set to None to disable warning detection.
+
+    max_byte_size: int, optional (default=33554432)
+        (`RegressionHoeffdingTree` parameter)
+        Maximum memory consumed by the tree.
+
+    memory_estimate_period: int, optional (default=2000000)
+        (`RegressionHoeffdingTree` parameter)
+        Number of instances between memory consumption checks.
+
+    grace_period: int, optional (default=50)
+        (`RegressionHoeffdingTree` parameter)
+        Number of instances a leaf should observe between split attempts.
+
+    split_confidence: float, optional (default=0.01)
+        (`RegressionHoeffdingTree` parameter)
+        Allowed error in split decision, a value closer to 0 takes longer
+        to decide.
+
+    tie_threshold: float, optional (default=0.05)
+        (`RegressionHoeffdingTree` parameter)
+        Threshold below which a split will be forced to break ties.
+
+    binary_split: bool, optional (default=False)
```

Comment:
The commonly used standard value for this parameter is `True`.

Reply:
Both RegressionHoeffdingTree & AdaptiveRandomForestClassifier use `False` as a default though, so I stuck with that. I think it's reasonable.

Reply:
Ok. My bad, I verified the Regression Hoeffding Tree implementation and you are right. I'm also going to resolve this point.